scripts/release/release.sh

```diff
@@ -0,0 +1,132 @@
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+#
+# A manual build script to build release and update github repo
+#
+# Display commands being run.
+# WARNING: please only enable 'set -x' if necessary for debugging, and be very
+#  careful if you handle credentials (e.g. from Keystore) with 'set -x':
+#  statements like "export VAR=$(cat /tmp/keystore/credentials)" will result in
+#  the credentials being printed in build logs.
+#  Additionally, recursive invocation with credentials as command-line
+#  parameters, will print the full command, with credentials, in the build logs.
+# set -x
+
+# Fail on any error.
+set -e
+# Fail on errors in pipes.
+set -o pipefail
+
+echo Enter client id:
+read -s -r GIT_CLIENT_ID
+echo Enter token:
+read -s -r GIT_CLIENT_TOKEN
+
+#Variables
+export SCRIPT_WORKING_DIR="${pwd}"
+export RELEASE_SCRIPT_DIR="$(dirname "$0")"
+export SCRIPT_PARENT_DIR="$(dirname -- "$RELEASE_SCRIPT_DIR")"
+export REPO_DIR="$(dirname -- "$SCRIPT_PARENT_DIR")"
+export RELEASE_OUTPUT_FILE="${REPO_DIR}/bazel-bin/dist/hadoop-migration-assessment-hooks.zip"
+export BUILD_SCRIPT="$RELEASE_SCRIPT_DIR/build.sh"
+source "${RELEASE_SCRIPT_DIR}/release_utils.sh"
+
+cd "${REPO_DIR}"
+
+export LAST_GIT_TAG="$(git tag |
+  grep -E '^v[0-9]' |
+  sort -V |
+  tail -1)"
+
+use_existing_tag=false
+
+# Do we already know what version we want to release?
+if [[ -z "${USE_TAG}" ]]; then
+  if [[ -z "${LAST_GIT_TAG}" ]]; then
+    err "No previous git tag found and it was not provided with USE_TAG env"
+  fi
+  VERSION="$(increment_tag_version ${LAST_GIT_TAG})"
+  log "Will create new version ${VERSION}"
+
+  if [ "$(git tag -l ${VERSION})" ]; then
+    err "ERROR! Tag for ${VERSION} already exists!"
+  fi
+
+  code="$(http_get_error_code "Accept: application/vnd.github.v3+json" \
+    "https://api.github.com/repos/google/hadoop-migration-assessment-tools/releases/tags/${VERSION}")"
+  if [ "${code}" != "404" ]; then
+    err "ERROR! Release with ${VERSION} tag version name already exists or http failed! Http code is ${code}"
+  fi
+else
+  VERSION="${USE_TAG}"
+  if [[ "${LAST_RELEASE_TAG}" ]]; then
+    LAST_GIT_TAG=${LAST_RELEASE_TAG}
+  fi
+  use_existing_tag=true
+  git checkout tags/"${VERSION}" -b "${VERSION}-release"
+fi
+
+log "Version name ${VERSION} verified"
+
+# Run build and integration tests
+cd "${SCRIPT_WORKING_DIR}"
+log "Build script : ${BUILD_SCRIPT}, currend dir $(pwd)"
+bash "${BUILD_SCRIPT}"
+
+# revert to initial state after running build script
+cd "${REPO_DIR}"
+
+# create and register tag for this release if it was not manually provided
+if ! [[ $use_existing_tag ]]; then
```

Comment:
The `use_existing_tag` variable is a string `true` or `false`. For the `if` condition to work, it cannot be added to `[[`, otherwise it will be always interpreted as true.

I would suggest changing to:
```
if ! "$use_existing_tag" ; then
```